### PR TITLE
Adding sample for Bulk support

### PR DESCRIPTION
**Pull request by ealsur**
# Pull Request Template

## Description

Adding sample usage project for Bulk support feature

### CODE REVIEW

File: Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+                {
+                    throw new ArgumentException("Please specify a valid AuthorizationKey in the appSettings.json");
+                }
+
+                CosmosClient client = Program.GetNormalClientInstance(endpoint, authKey);
+                CosmosClient bulkClient = Program.GetBulkClientInstance(endpoint, authKey);
+                // Create the require container, can be done with any client
+                await Program.InitializeAsync(client);
+
+                Console.WriteLine("Running demo with a normal CosmosClient...");
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs]: Typical usage will only cover the target scenario not comparison.

File: Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                for (int i = 0; i < Program.concurrentDocuments; i++)
+                {
+                    string id = Guid.NewGuid().ToString();
+                    MyDocument myDocument = new MyDocument() { id = id, pk = partitionKeyValue };
+                    tasks.Add(container.CreateItemAsync<MyDocument>(myDocument, new PartitionKey(partitionKeyValue)));
+                }
+
+                while (tasks.Count > 0)
+                {
+                    Task<ItemResponse<MyDocument>> t = await Task.WhenAny(tasks);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs]: How about WhenAll? Will that simplify the logic?

File: Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                    try {
+                        ItemResponse<MyDocument> r = await t;
+                        if (r.StatusCode == HttpStatusCode.Created)
+                        {
+                            itemsCreated++;
+                        }
+                    }
+                    catch(CosmosException ex) when (ex.StatusCode == HttpStatusCode.TooManyRequests)
+                    {
+                        // With a high volume of operations, some throttling might occur, in this case, we do not consider them successful responses
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs]: Would be nice to print the item details (id, PK).
It a template for applications on how to handler any other errors also.

File: Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs (hunk 1 above)
Comment (ealsur) [reply Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs]: Removed

File: Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs (hunk 2 above)
Comment (ealsur) [reply Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs]: Added WhenAll

File: Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs (hunk 3 above)
Comment (ealsur) [reply Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs]: Added printing through ContinueWith

File: Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+                {
+                    string id = Guid.NewGuid().ToString();
+                    MyDocument myDocument = new MyDocument() { id = id, pk = partitionKeyValue };
+                    tasks.Add(
+                        container.CreateItemAsync<MyDocument>(myDocument, new PartitionKey(partitionKeyValue))
+                        .ContinueWith((Task<ItemResponse<MyDocument>> task) =>
+                        {
+                            if (task.IsCompletedSuccessfully)
+                            {
+                                itemsCreated++;
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs]: What's the significance of updating local variable?

Comment (ealsur) [reply Microsoft.Azure.Cosmos.Samples/Usage/BulkSupport/Program.cs]: Good catch

**Pull request merged**